src/core/ext/transport/chttp2/transport/frame_rst_stream.c

```diff
@@ -103,6 +103,9 @@ grpc_error *grpc_chttp2_rst_stream_parser_parse(grpc_exec_ctx *exec_ctx,
           GRPC_ERROR_INT_HTTP2_ERROR, (intptr_t)reason);
       gpr_free(message);
     }
+    if (!s->write_closed) {
```

Comment:
Same question. Seems like this will occur from [here](https://github.com/grpc/grpc/blob/master/src/core/ext/transport/chttp2/transport/chttp2_transport.c#L2103)